placesautocomplete/src/main/java/com/seatgeek/placesautocomplete/json/AndroidPlacesApiJsonParser.java

```diff
@@ -1,32 +1,844 @@
 package com.seatgeek.placesautocomplete.json;
 
+import android.util.JsonReader;
+import android.util.JsonWriter;
+
+import com.seatgeek.placesautocomplete.model.AddressComponent;
+import com.seatgeek.placesautocomplete.model.AddressComponentType;
+import com.seatgeek.placesautocomplete.model.AlternativePlaceId;
+import com.seatgeek.placesautocomplete.model.DateTimePair;
+import com.seatgeek.placesautocomplete.model.DescriptionTerm;
+import com.seatgeek.placesautocomplete.model.MatchedSubstring;
+import com.seatgeek.placesautocomplete.model.OpenHours;
+import com.seatgeek.placesautocomplete.model.OpenPeriod;
 import com.seatgeek.placesautocomplete.model.Place;
+import com.seatgeek.placesautocomplete.model.PlaceDetails;
+import com.seatgeek.placesautocomplete.model.PlaceGeometry;
+import com.seatgeek.placesautocomplete.model.PlaceLocation;
+import com.seatgeek.placesautocomplete.model.PlacePhoto;
+import com.seatgeek.placesautocomplete.model.PlaceReview;
+import com.seatgeek.placesautocomplete.model.PlaceScope;
+import com.seatgeek.placesautocomplete.model.PlaceType;
 import com.seatgeek.placesautocomplete.model.PlacesAutocompleteResponse;
 import com.seatgeek.placesautocomplete.model.PlacesDetailsResponse;
+import com.seatgeek.placesautocomplete.model.RatingAspect;
+import com.seatgeek.placesautocomplete.model.Status;
 
+import java.io.BufferedReader;
+import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.util.ArrayList;
 import java.util.List;
 
 class AndroidPlacesApiJsonParser implements PlacesApiJsonParser {
 
     @Override
-    public PlacesAutocompleteResponse autocompleteFromStream(final InputStream is) {
-        throw new UnsupportedOperationException("Not yet implemented, please have Gson on your classpath");
+    public PlacesAutocompleteResponse autocompleteFromStream(final InputStream is) throws JsonParsingException {
+        try {
```

Comment:
Also, in some places, you're calling `reader.close()` and others you're not. If it needs to be called, I'd prefer it in the finally block as well.